staging/src/kubevirt.io/client-go/kubecli/vm.go

```diff
@@ -150,3 +153,17 @@ func (v *vm) Stop(name string) error {
 	uri := fmt.Sprintf(vmSubresourceURL, v1.ApiStorageVersion, v.namespace, name, "stop")
 	return v.restClient.Put().RequestURI(uri).Do().Error()
 }
+
+func (v *vm) Migrate(name string) error {
```

Comment:
I do not sure it is the correct way, in this case, we do not have an endpoint for migration under subresources, it maybe not critical, but nice to have, so the plan:
- implement migration point under subresources
- call to it similar to all other commands

Reply:
That's also possible, maybe even better as other clients could trigger the migration by invoking the migrate operation on the VM. But I think such a new endpoint would render the existing one redundant - and dropping it is a bigger and more controversial change I believe.

The propose change provides the ability to migrate the VM through `virtctl` and limits the modifications only to `virtctl`. How about getting proceeding with it and consider changing/adding the endpoint you propose separately? it would be fairly quick to change `virtctl` to cope with it. In the meantime having the `migrate` functionality can be helpful to users as well as for us for debugging

Reply:
will work for me, I will not stop this PR on it, and will wait for the following one:)

Reply:
I think Artyom is right: I don't remember the reason (maybe @fabiand or @mfranczy do?), but the other virtctl commands use subresources, and I think we should not change this.

> But I think such a new endpoint would render the existing one redundant

which existing one?

Reply:
The reason is simply that for every thing we wanted to do conveniently, the work was don on the server side first, and then exposed via virtctl. To avoid that we build up technical debt.

Reply:
> > But I think such a new endpoint would render the existing one redundant
> 
> which existing one?

Please ignore that part. Talked with Artyom offline earlier and realized I mixed two unrelated things here

Reply:
@cynepco3hahue @slintes this seems to require enabling kubevirt-apiserver to create resources on `virtualmachineinstancemigrations.kubevirt.io`, would it be ok?

Reply:
@ahadas yes

Reply:
Done